### ISSUE TITLE
responds_to? requires using `self` inside the conditional, but probably shouldn't

### ISSUE THREAD

**Issue by paulcsmith** [crystal-lang/crystal#5612]
Crystal 0.24.1
MacOS High Sierra 10.13.2

https://play.crystal-lang.org/#/r/3ezs

```
class Users::IndexPage
  def render
    if responds_to?(:extra_content)
      extra_content
      # Works if you do self.extra_content
    end
  end
end

Users::IndexPage.new.render

# Produces: Error in line 9: instantiating 'Users::IndexPage#render()' in line 4: undefined local variable or method 'extra_content'
```

### Possible solutions

1. Compiler could figure out that it is a method call without `self.`
1. The error could be improved so that if you make a method call with `self.` inside of a conditional with `responds_to?` it will tell you to add `self.`. Otherwise, it's pretty hard to figure out what to do.

Any other ideas?

**Comment by asterite** [comment]
Probably `responds_to?` shouldn't work for implicit self because of this reason. Maybe it even shouldn't work with `self` and you'll have to assign it to a different variable first. It's pretty confusing.

**Comment by paulcsmith** [comment]
Yeah it's definitely confusing. I'm fine with self.responds_to?, but I think there should be a nice error message that makes it clear what Crystal is expecting. Otherwise, I think people will be pretty stumped by it
